addons/account/wizard/account_invoice_send.py

```diff
@@ -173,3 +173,11 @@ def save_as_template(self):
         action = _reopen(self, self.id, self.model, context=self._context)
         action.update({'name': _('Send Invoice')})
         return action
+
+class MailComposer(models.TransientModel):
+    _inherit = 'mail.compose.message'
+
+    def action_send_mail(self):
+        """ Used for action button that do not accept arguments. """
+        self.with_context(mailing_document_based=True)._action_send_mail(auto_commit=False)
+        return {'type': 'ir.actions.act_window_close'}
```

Comment:
This just adds mailing_document_based on all composers when this module is installed again. Not ok.

Reply:
also side note if it was ok (and it's not), it's better if the override is minimal so something like:

```py
def action_send_mail(self):
    return super(MailComposer, self.with_context(mailing_document_based=True)).action_send_mail(self)
```